src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -216,6 +216,8 @@ private static void WriteClientMethod(CodeWriter writer, LowLevelRestClient clie
 
             using (WriteClientMethodDeclaration(writer, restMethod, clientMethod.OperationSchemas, returnType, async))
             {
+                writer.WriteParameterNullChecks(clientMethod.RestMethod.Parameters);
```

Comment:
Where do we check whether the parameter is required or optional?

Reply:
My understanding is as follows: parameter objects have the bool property `ValidateNotNull`, which is used [here](https://github.com/Azure/autorest.csharp/blob/5474a4124a0e64707779e9612e6da4c3fbd81e24/src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs#L213) to decide whether to write the code check for this variable or not. Default values for the parameter also seem to be taken into consideration.

Am I answering the right question? Or are you asking about how the properties are set in the first place?